app/src/lib/Map.svelte

```diff
@@ -50,8 +50,32 @@
         return div
     }
 
+    const initializeUserTracking = () => {
+        let userLocation = setInterval(() => {
+            navigator.geolocation.getCurrentPosition((pos) => {
+                icons.push(L.marker([pos.coords.latitude, pos.coords.longitude])
+                    .addTo(map));
+            },
+            (err) => {
+                console.warn(`ERROR(${err.code}): ${err.message}`);
+            },
+            {
+                enableHighAccuracy: true,
+                timeout: 5000,
+                maximumAge: 0
+            });
+        }, 2000);
+        
+        return {
+            destroy: () =>
+                clearInterval(userLocation)
+        }
+    }
```

Comment:
@MarcusDunn does this make sense for clearing the interval?

Reply:
almost. you have to call it in the same function that `map.remove()` is in.

Reply:
change `initializeUserTracking();` to `const stopTrackingUser = initializeUserTracking();`

and change ```return {
            destroy: () => {
                map.remove()
            },
        };```

to ```
return {
            destroy: () => {
                map.remove()
                stopTrackingUser()
            },
        };```

Reply:
So do I need to remove the `destroy `from `initializeUserTracking`?

Reply:
if you *only* have it return a function that calls `clearInterval` I think that would be simplest. you could also do (keeping the return as it is now) to `stopTrackingUser.destroy()`